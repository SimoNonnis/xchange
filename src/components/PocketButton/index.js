import React from 'react';
import PropTypes from 'prop-types';

const propTypes = {
  code: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  symbol: PropTypes.string.isRequired,
  amount: PropTypes.number.isRequired,
  isSelected: PropTypes.bool.isRequired,
  onClick: PropTypes.func
};

const defaultProps = {
  onClick: () => undefined
};

const PocketButton = ({ code, name, symbol, amount, isSelected, onClick }) => (
  <button
    className={`buttonPocket u-fontSizeSmall ${
      isSelected ? '' : 'u-semiTransparent'
    }`}
    disabled={isSelected}
    onClick={onClick}>
    <div className={`${isSelected ? 'isSelectedPocket' : ''}`}>
      <span>{symbol}</span>
      <span>{amount}</span>
    </div>
    {isSelected && (
      <div className="buttonPocketInfo">
        <span>{code}</span> - <span>{name}</span>
      </div>
    )}
  </button>
);

PocketButton.propTypes = propTypes;
PocketButton.defaultProps = defaultProps;

export default PocketButton;
