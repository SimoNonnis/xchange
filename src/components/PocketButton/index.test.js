import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import PocketButton from './index';

describe('Test <PocketButton />', () => {
  it('should render with required props and selected = false', () => {
    const wrapper = shallow(
      <PocketButton
        code="RUB"
        name="Russian Ruble"
        symbol="₽"
        amount={1000}
        isSelected={false}
      />
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render with required props and selected = true', () => {
    const wrapper = shallow(
      <PocketButton
        code="RUB"
        name="Russian Ruble"
        symbol="₽"
        amount={1000}
        isSelected={true}
      />
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render with all props', () => {
    const onClick = jest.fn();
    const wrapper = shallow(
      <PocketButton
        code="RUB"
        name="Russian Ruble"
        symbol="₽"
        amount={1000}
        isSelected={false}
        onClick={onClick}
      />
    );

    wrapper.simulate('click');
    expect(onClick).toHaveBeenCalled();
  });
});
