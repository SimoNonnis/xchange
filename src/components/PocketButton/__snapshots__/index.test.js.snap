// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <PocketButton /> should render with required props and selected = false 1`] = `
<button
  className="buttonPocket u-fontSizeSmall u-semiTransparent"
  disabled={false}
  onClick={[Function]}
>
  <div
    className=""
  >
    <span>
      ₽
    </span>
    <span>
      1000
    </span>
  </div>
</button>
`;

exports[`Test <PocketButton /> should render with required props and selected = true 1`] = `
<button
  className="buttonPocket u-fontSizeSmall "
  disabled={true}
  onClick={[Function]}
>
  <div
    className="isSelectedPocket"
  >
    <span>
      ₽
    </span>
    <span>
      1000
    </span>
  </div>
  <div
    className="buttonPocketInfo"
  >
    <span>
      RUB
    </span>
     - 
    <span>
      Russian Ruble
    </span>
  </div>
</button>
`;
